### PR TITLE
8249451: Unconditional exceptions clearing logic in compiler code should honor Async Exceptions.

### PR DESCRIPTION
**Pull request by jamsheedcm**
Hi

Moving the review that is based on mercurial repo to github.
The history of conversation is [here](https://mail.openjdk.java.net/pipermail/hotspot-compiler-dev/2020-September/039861.html)

Issue:[ JDK-8249451 ](https://bugs.openjdk.java.net/browse/JDK-8249451)

@dholmes-ora could you please have a look.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8249451](https://bugs.openjdk.java.net/browse/JDK-8249451): Unconditional exceptions clearing logic in compiler code should honor Async Exceptions.

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 1c0786a59fcd8d728778f41cdc1917df98d070bd
 * [Igor Veresov](https://openjdk.java.net/census#iveresov) (@veresov - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/169/head:pull/169`
`$ git checkout pull/169`

### CODE REVIEW

File: src/hotspot/share/runtime/thread.cpp
Comment (dholmes-ora) [inline src/hotspot/share/runtime/thread.cpp:2392]: Suggest:
// We may be at method entry which requires we save the do-not-unlock flag.

**Review by dholmes-ora — APPROVED** [review]
Looks good to me! No further comments.

File: src/hotspot/share/runtime/thread.cpp
Comment (jamsheedcm) [reply src/hotspot/share/runtime/thread.cpp]: Thank you @dholmes-ora , Done.

**Comment by jamsheedcm** [comment]
could you i get second review

File: src/hotspot/share/prims/whitebox.cpp
Comment (jamsheedcm) [reply src/hotspot/share/prims/whitebox.cpp:1056]: bool WhiteBox::compile_method returns a bool

Comment (veresov) [reply src/hotspot/share/prims/whitebox.cpp:1056]: Yes, you're right of course. Reviewed.

**Review by veresov — APPROVED** [review]

**Review by veresov — COMMENTED** [review]
Looks good.

**Pull request closed without merging**
